discovery/file/file.go

```diff
@@ -240,12 +317,21 @@ func fileSource(filename string, i int) string {
 
 // readFile reads a JSON or YAML list of targets groups from the file, depending on its
 // file extension. It returns full configuration target groups.
-func readFile(filename string) ([]*config.TargetGroup, error) {
+func readFile(index int, filename string) ([]*config.TargetGroup, error) {
 	content, err := ioutil.ReadFile(filename)
 	if err != nil {
 		return nil, err
 	}
 
+	// We use a mutex when setting timestamp, so don't block the rest of readFile.
```

Comment:
You'd be better off reading the file using the `os` package and calling `File.Stat`, that way you know it' the timestamp of the file you read.

Reply:
`ioutil.readFile` uses `os.Open`, so I think we're safe here

https://golang.org/src/io/ioutil/ioutil.go?s=1464:1510#L39

Reply:
ah yeah you're right, I see what you mean now